### PR TITLE
[Exporter.Geneva] Add data validation for MetricExportIntervalMilliseconds

### PR DESCRIPTION
**Pull request by Yun-Ting**
Fixes https://github.com/open-telemetry/opentelemetry-dotnet-contrib/issues/394.

### CODE REVIEW

File: src/OpenTelemetry.Exporter.Geneva/GenevaMetricExporterOptions.cs [hunk 1]
```diff
@@ -34,7 +35,28 @@ public class GenevaMetricExporterOptions
...
+        get
+        {
+            return this._metricExporterIntervalMilliseconds;
+        }
+
+        set
+        {
+            if (value < 0)
+            {
+                throw new ArgumentException($"MetricExportIntervalMilliseconds: {value} should be a greater than 0 integer.");
```
Comment (Kielek) [inline src/OpenTelemetry.Exporter.Geneva/GenevaMetricExporterOptions.cs:49]: Consider to use `ArgumentOutOfRangeException`. It inherits from `ArgumentException` but it is more descriptive.

Comment (reyang) [reply src/OpenTelemetry.Exporter.Geneva/GenevaMetricExporterOptions.cs:49]: Follow https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/dcf0156af8697c0f4d95e4fdae8d3b4b778e4810/src/OpenTelemetry.Extensions.AzureMonitor/ApplicationInsightsSampler.cs#L37.

File: test/OpenTelemetry.Exporter.Geneva.Tests/GenevaMetricExporterOptionsTests.cs [hunk 2]
```diff
@@ -69,5 +69,24 @@ public void InvalidPrepopulatedDimensions()
...
+                    MetricExportIntervalMilliseconds = 500,
+                };
+            });
+
+#if NET6_0 || NETCOREAPP3_1_OR_GREATER
+            var negativeInputExpectedErrorMessage = "Must be in the range: [1000, 2147483647] (Parameter 'value')\r\nActual value was 500.";
+#else
+            var negativeInputExpectedErrorMessage = "Must be in the range: [1000, 2147483647]\r\nParameter name: value\r\nActual value was 500.";
+#endif
+            Assert.Equal(negativeInputExpectedErrorMessage, negativeInputMetricExportIntervalMillisecondsException.Message);
```
Comment (reyang) [inline test/OpenTelemetry.Exporter.Geneva.Tests/GenevaMetricExporterOptionsTests.cs:89]: Do we really want this? My gut feeling is that the actual error message is owned by `Guard`, and if we really want to validate it, it should be in `GuardTests.cs`.

BTW, the use of EOL (CRLF vs. LF) might break depending on the operating system.

Comment (Yun-Ting) [reply test/OpenTelemetry.Exporter.Geneva.Tests/GenevaMetricExporterOptionsTests.cs]: So maybe only validating whether it's an ArgumentOutofRange sufficient here?
I am not sure whether we really need this test.

**Review by reyang — APPROVED** [review]

File: test/OpenTelemetry.Exporter.Geneva.Tests/GenevaMetricExporterOptionsTests.cs (hunk 2 above)
Comment (utpilla) [reply test/OpenTelemetry.Exporter.Geneva.Tests/GenevaMetricExporterOptionsTests.cs]: I think we should definitely test that we throw an `ArgumentOutOfRangeException` when we provide a value less than 1000 ms.

@reyang How do we test that we have called the right `Guard` method with the right set of parameters? I think we could assert the message here as well to ensure that.

Comment (reyang) [reply test/OpenTelemetry.Exporter.Geneva.Tests/GenevaMetricExporterOptionsTests.cs]: I think the following tests should be good enough:
1. use 999ms, the ctor should complain
2. use 1000ms, the ctor should succeed

File: test/OpenTelemetry.Exporter.Geneva.Tests/GenevaMetricExporterOptionsTests.cs [hunk 3]
```diff
@@ -69,5 +69,27 @@ public void InvalidPrepopulatedDimensions()
...
         }
+
+        [Fact]
+        public void MetricExportIntervalValidationTest()
+        {
+            Assert.Throws<ArgumentOutOfRangeException>(() =>
+            {
+                var exporterOptions = new GenevaMetricExporterOptions
+                {
+                    MetricExportIntervalMilliseconds = 500,
```
Comment (reyang) [inline test/OpenTelemetry.Exporter.Geneva.Tests/GenevaMetricExporterOptionsTests.cs:80]: Minor - consider testing on the boundary:

```suggestion
                    MetricExportIntervalMilliseconds = 999,
```

**Review by utpilla — APPROVED** [review]

File: test/OpenTelemetry.Exporter.Geneva.Tests/GenevaMetricExporterOptionsTests.cs (hunk 3 above)
Comment (utpilla) [reply test/OpenTelemetry.Exporter.Geneva.Tests/GenevaMetricExporterOptionsTests.cs]: @Yun-Ting I have committed this suggestion.

**Review by Kielek — APPROVED** [review]

**Review by cijothomas — APPROVED** [review]

**Pull request merged**